src/ts/components/timelinemarkershandler.ts

```diff
@@ -121,20 +121,32 @@ export class TimelineMarkersHandler {
         .map(cssClass => this.prefixCss(cssClass));
 
       const cssProperties: { [propertyName: string]: string } = {
-        'width': marker.position + '%',
+        'left': `${marker.position}%`,
       };
 
       if (marker.duration > 0) {
         const markerWidthPx = Math.round(seekBarWidthPx / 100 * marker.duration);
-        cssProperties['border-right-width'] = markerWidthPx + 'px';
-        cssProperties['margin-left'] = '0';
+        cssProperties['width'] = `${markerWidthPx}px`;
       }
 
-      this.markersContainer.append(new DOM('div', {
+      const markerElement = new DOM('div', {
         'class': markerClasses.join(' '),
         'data-marker-time': String(marker.marker.time),
         'data-marker-title': String(marker.marker.title),
-      }).css(cssProperties));
+      }).css(cssProperties);
+
+      if (marker.marker.imageUrl) {
+        const markerImageClasses = ['seekbar-marker-image'].map(cssClass => this.prefixCss(cssClass));
+
+        const imageElement = new DOM('img', {
+          'class': markerImageClasses.join(' '),
```

Comment:
Why do you create and map the array with `markerImageClass` only to join the single array element then again?

Reply:
Fixed in https://github.com/bitmovin/bitmovin-player-ui/pull/318/commits/7bdc58a429002869096c7d3f7eeee5ae36b7d22b